nemo/collections/nlp/modules/common/megatron/token_level_encoder_decoder.py

```diff
@@ -210,72 +174,116 @@ def __init__(
                     # We initialize decoder embeddings, but set them to zero since we they're tied with the encoder embeddings.
                     # A later initialize_embedding call will synchronize the embeddings.
                     self.decoder_embedding = Embedding(
-                        hidden_size=hidden_size,
+                        hidden_size=decoder_cfg.hidden_size,
```

Comment:
Need to make sure merging of untied encoder-decoder embedding is merged correctly here

Reply:
Yes exactly. Which is why I'll let you merge first and then fix it in this PR.

Reply:
I also need sequence parallel to be merged to fix RPE related things

Reply:
Done!